Lib/test/test_structseq.py

```diff
@@ -133,6 +135,74 @@ def test_match_args_with_unnamed_fields(self):
         self.assertEqual(os.stat_result.n_unnamed_fields, 3)
         self.assertEqual(os.stat_result.__match_args__, expected_args)
 
+    def test_copy_replace_all_fields_visible(self):
+        assert os.times_result.n_unnamed_fields == 0 and os.times_result.n_sequence_fields == os.times_result.n_fields
+
+        expected_args = ('user', 'system', 'children_user', 'children_system', 'elapsed')
+        self.assertEqual(os.times_result.__match_args__, expected_args)
+        self.assertEqual(os.times_result.n_fields, len(expected_args))
+
+        n_fields = os.times_result.n_fields  # 5
+        t = os.times_result(range(os.times_result.n_fields))
+
+        # visible fields
+        self.assertEqual(copy.replace(t), tuple(range(n_fields)))
+        self.assertEqual(copy.replace(t, user=1), (1, *range(1, n_fields)))
+        self.assertEqual(copy.replace(t, system=2), (0, 2, *range(2, n_fields)))
+        self.assertEqual(copy.replace(t, user=1, system=2), (1, 2, *range(2, n_fields)))
```

Comment:
Few suggestions:
* Test that the result has the correct type, not simple tuple.
* Use float values. Currently there are no restrictions, but normally they are floats.
* Use subscription and slices to get original field values.

```suggestion
        self.assertEqual(copy.replace(t), t)
        self.assertIsInstance(copy.replace(t), os.times_result)
        self.assertEqual(copy.replace(t, user=1.5), (1.5, *t[1:]))
        self.assertEqual(copy.replace(t, system=2.5), (t[0], 2.5, *t[2:]))
        self.assertEqual(copy.replace(t, user=1.5, system=2.5), (1.5, 2.5, *t[2:]))
```